### PR TITLE
Create the DNS record only when hosted zone exits

### LINKED ISSUE #632: RRSet with DNS name foo.vercel.com. is not permitted in zone next.com.
### Describe the bug

Not sure if it's a bug

I have trouble adding domain from another AWS account, i have been setting it manually all this time, i get this error `RRSet with DNS name foo.vercel.com. is not permitted in zone next.com.`

Vercel and Next are just example

AWS Account 1: vercel.com
AWS Account 2: next.com (the account that i host the nextjs app)

Created ACM certificate from Account 2 with these details and approved
domain name: vercel.com
additional name: foo.vercel.com

serverless.yml
```yaml
   inputs:
        domain:
            - foo
            - vercel.com
```

when i'm deploying i get the above error, not sure what i can do.

Currently i set the CNAME and Certificate manually in AWS Console

If it's not a bug, is it possible to allow passing CNAME and Viewer certificate to cloudfront inputs?

### Versions

- OS/Environment: MacOS
- @sls-next/serverless-component version: 1.17

### Checklist

- [x] You have reviewed the README and [FAQs](https://github.com/serverless-nextjs/serverless-next.js#faq), which answers several common questions.
- [x] Consider using the latest [release version](https://github.com/serverless-nextjs/serverless-next.js/releases), which may have already fixed your issue.

### PR DESCRIPTION
**Pull request by SarKurd**
Fixes the DNS creation error where hosted zone is configured in a different account.

closes #632

### CODE REVIEW

File: packages/serverless-components/domain/serverless.js [hunk 1]
```diff
@@ -129,14 +129,20 @@ class Domain extends Component {
...
+            clients.route53,
+            subdomain,
+            domainHostedZoneId,
+            subdomain.url.replace("https://", ""),
+            inputs.domainType,
+            this
+          );
+        } else {
+          this.context.debug(
+            `DNS recods for domain "${subdomain.domain}" were not configured because the domain was not found in your account. Please configure the DNS records manually`
```
Comment (dphang) [inline packages/serverless-components/domain/serverless.js:143]: typo: "DNS recods" -> "DNS records"

**Review by dphang — COMMENTED** [review]
Is it possible to add some small unit test for this?

File: packages/serverless-components/domain/serverless.js [hunk 2]
```diff
@@ -191,15 +197,17 @@ class Domain extends Component {
...
+          this.context.debug(
+            `Removing CloudFront DNS records for domain ${domainState.domain}`
+          );
+          await removeCloudFrontDomainDnsRecords(
+            clients.route53,
+            domainState.domain,
+            domainHostedZoneId,
+            domainState.url.replace("https://", "")
+          );
+        }
```
Comment (dphang) [inline packages/serverless-components/domain/serverless.js:210]: Not sure if it's useful to have a message in the `else` case too?

**Comment by SarKurd** [comment]
> Is it possible to add some small unit test for this?

There are no tests at all for Domain component that's why i didn't create any, if you don't mind i'll open another PR in the next few days and add unit tests for the whole domain component

File: packages/serverless-components/domain/serverless.js (hunk 2 above)
Comment (SarKurd) [reply packages/serverless-components/domain/serverless.js:210]: User by now knows there is no DNS record to be removed but i'll add one

**Comment by dphang** [comment]
> > Is it possible to add some small unit test for this?
> 
> There are no tests at all for Domain component that's why i didn't create any, if you don't mind i'll open another PR in the next few days and add unit tests for the whole domain component

Oh yeah, that's fine for now then. We can always improve it later. The e2e tests aren't validating the `domain` input, so just wanted to be a bit safer. As long as you tested, I am fine. Thanks!

**Comment by dphang** [comment]
Looks like e2e tests are failing but that's because PRs of forks cannot access the AWS access keys used for it (for security reasons, of course). I'll merge it anyway since other tests pass and fix this later.

**Review by dphang — COMMENTED** [review]
Approved!

**Pull request merged**